commit 59337087cb33db58aa0d4463892b4475cf66a50b
tree 10fe5d789fd08fab0b23e0b152397afed89862d3
parent 19f00e66f8aa7ee581c6d003fd68ee9f9dee4057
author Stefan Richter <stefanr@s5r6.in-berlin.de> 1183583633 +0200
committer Stefan Richter <stefanr@s5r6.in-berlin.de> 1184018859 +0200

    ieee1394: raw1394: fix a 32/64-bits compat fix
    
    I was told that only i386 aligns 64 bit integers at 4 bytes boundaries
    while all other architectures (32 bit architectures with 64 bit
    siblings) align it on 8 bytes boundaries.
    
    Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>

---
 drivers/ieee1394/raw1394.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/ieee1394/raw1394.c	2007-08-27 14:01:24.000000000 -0400
+++ b/drivers/ieee1394/raw1394.c	2007-08-27 14:01:24.000000000 -0400
@@ -434,7 +434,11 @@ struct compat_raw1394_req {
 
 	__u64 sendb;
 	__u64 recvb;
-} __attribute__((packed));
+}
+#if defined(CONFIG_X86_64) || defined(CONFIG_IA64)
+__attribute__((packed))
+#endif
+;
 
 static const char __user *raw1394_compat_write(const char __user *buf)
 {
